### ISSUE TITLE
rust-analyzer: requires cargo in path to work properly

### ISSUE THREAD

**Issue by buffet** [NixOS/nixpkgs#97926]
**Describe the bug**

Rust analyzer needs cargo to work properly, otherwise it says "Rust Analyzer is loading..." infinitely

**To Reproduce**
1. install rust-analyzer
2. make sure cargo isn't in your path
3. try to use it

**Expected behavior**

It should just work.

**Notify maintainers**

@oxalica 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

- system: `"x86_64-linux"`
 - host os: `Linux 5.4.58, NixOS, 20.03post-git (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - nixpkgs: `/nix/store/96pn32ivdpxaz8d8cq765jhllvb334xl-source`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: rust-analyzer
```

**Comment by oxalica** [comment]
`rust-analyzer` is packaged to use cargo from PATH intentionally (which is also the default behavior), since users usually have `rustup` or `<mozilla-overlay>` for more up-to-date and/or customized rust toolchains, instead of locked `cargo` from nixpkgs itself.

For the error message, it seems to be already fixed by #97444. I think it's clear enough:

<img width="343" alt="image" src="https://user-images.githubusercontent.com/14816024/93079137-3c0e2300-f6be-11ea-8436-9fab59fcfc6b.png">

**Comment by buffet** [comment]
Alright, makes sense. Thanks for the fast reply.
